### PR TITLE
feat: calculate distances

### PR DESCRIPTION
**Pull request by codeFactory12**
## Type of Change
Implements the calcule for the distance for start to end points.

- [x ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by MorveN11 — COMMENTED** [review]
Please check all of these comments before merging your branch. for any doubt feel free to contact me.

File: apps/client/src/app/pages/map/hooks/useDistanceByBike.tsx
Comment (MorveN11) [inline apps/client/src/app/pages/map/hooks/useDistanceByBike.tsx:1]: This is not a hook for all the map, please move these files to /app/pages/map/components/route/hooks/*

**Review by MorveN11 — COMMENTED** [review]
Please check all of these comments before merging your branch. for any doubt feel free to contact me.

File: apps/client/src/app/pages/map/hooks/useDistanceByBike.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      start.lng
+    };${end.lat},${end.lng}?access_token=${
+      import.meta.env.VITE_PUBLIC_MAPBOX_TOKEN
+    }`
+  );
+
+  const data = (await response.json()) as DirectionsResponse;
+
+  const { distance } = data.routes[0];
+  const distanceInKilometers = Math.round((distance / 1000) * 100) / 100;
```
Comment (Severaldog) [inline apps/client/src/app/pages/map/hooks/useDistanceByBike.tsx:24]: Add equations to a variable with a descriptive name to make it possible to know what the equations do and make it easier to read.

**Review by Severaldog — APPROVED** [review]

**Pull request closed without merging**